feat: export pop command (#262)
babylonlabs-io/finality-provider: docs/pop_format_spec.md

# Proof of Possession (PoP) Specification

## Overview

> The Proof of Possession (PoP) mechanism verifies that the owner of an EOTS key
> pair also controls a BABY key pair. This validation requires five essential
> attributes, which are exposed by the `PoPExport` structure.

Reviewer: ```suggestion
The Proof of Possession (PoP) structured specification outlined in this document
allows for the verification of the mutual ownership of a Babylon key pair and
an EOTS key pair.
In the following,
we outline the five essential attributes exposed by the `PoPExport` structure
and provide examples and validation procedures.
```

## Attributes

The `PoPExport` structure is defined

> [here](https://github.com/babylonlabs-io/finality-provider/blob/cc07bcd4dc434f7095668724aad6865bffe425e0/eotsmanager/cmd/eotsd/daemon/pop.go#L36).

Reviewer: This links to a specific commit in the same repository,
so this will not transition well to future updates.
However, doing a relative reference will probably lead to
an outdated line reference in the future.
We could just copy the structure with its comments here maybe.

Your call

Author: pasted the struct

It contains the following fields:

> - `EotsPublicKey` – The EOTS public key of the finality provider, marshaled as
> a hexadecimal string. This value represents the public component of the EOTS
> key pair used in the signing process.

Reviewer: ```suggestion
- `EotsPublicKey`: The EOTS public key of the finality provider in hexadecimal format.
```

> - `BabyPublicKey` – The secp256k1 public key, marshaled as a base64 string.

Reviewer: ```suggestion
- `BabyPublicKey` – The Babylon secp256k1 public key in base64 format.
```

Very weird that we use base64 here, while on EOTS we use hex.

> This key is extracted from the BABY keyring and uniquely identifies the BABY

Reviewer: Let's use "Babylon" instead of "BABY" (across the document). "BABY" is the ticker. Ok to use it in attributes, e.g.  `BabyAddress`

key pair.

> - `BabyAddress` – The BABY address, prefixed with `bbn`. It is derived from the

Reviewer: ```suggestion
- `BabyAddress` – The Babylon account address (`bbn` prefix). The address is derived from the
```

`BabyPublicKey` and used as the primary identifier on the Babylon network.

> - `EotsSignBaby` – A Schnorr signature, created by signing the
> `sha256(BabyAddress)` with the private key corresponding to the EOTS public
> key. Encoded in base64, this ensures that the EOTS key can verify ownership
> of the BABY address.

Reviewer: ```suggestion
- `EotsSignBaby` – A Schnorr signature in base64 format, created by signing the
`sha256(BabyAddress)` with the EOTS private key.
```

- `BabySignEotsPk` – A signature of the `EotsPublicKey`, created by the BABY
private key. This signature follows the Cosmos
[ADR-036](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-036-arbitrary-signature.md)

> specification and is encoded in base64. It demonstrates that the BABY key pair
> acknowledges and signs the EOTS public key.

Reviewer: ```suggestion
specification and is encoded in base64.
```
